<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>satuRn - vignette • satuRn</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="satuRn - vignette">
<meta property="og:description" content="satuRn">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">satuRn</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette.html">satuRn - vignette</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jgilis/satuRn/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Vignette_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>satuRn - vignette</h1>
                        <h4 class="author">Jeroen Gilis</h4>
            
            <h4 class="date">30/11/2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jgilis/satuRn/blob/master/vignettes/Vignette.Rmd"><code>vignettes/Vignette.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette.Rmd</code></div>

    </div>

    
    
<p><code>satuRn</code> is an R package to perform differential transcript usage analyses in bulk and single-cell transcriptomics datasets. The package has three main functions. The first function, <code>fitDTU</code>, is used to model transcript usage profiles by means of a quasi-binomial generalized linear model. Second, the <code>testDTU</code> function tests for differential usage of transcripts between certain groups of interest (e.g. different treatment groups or cell types). Finally, the <code>plotDTU</code> can be used to visualize the usage profiles of selected transcripts in different groups of interest. All details about the <code>satuRn</code> model and statistical tests are described in our preprint […].</p>
<p>In this vignette, we analyze a small subset of the data from <span class="citation">(Tasic Bosiljka 2018)</span>. More specifically, an expression matrix and the corresponding metadata of the subset data has been provided with the <code>satuRn</code> package. We will adopt this dataset to showcase the different functionalities of <code>satuRn</code>.</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>For the moment, <code>satuRn</code> is only available from our GitHub page <a href="https://github.com/statOmics/satuRn" class="uri">https://github.com/statOmics/satuRn</a>. We hope to make a submission to the Bioconductor project soon.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#devtools::install_github("statOmics/satuRn")</span>
<span class="co">#test when no longer private</span></code></pre></div>
</div>
<div id="load-libraries" class="section level1">
<h1 class="hasAnchor">
<a href="#load-libraries" class="anchor"></a>Load libraries</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jgilis/satuRn">satuRn</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jorainer/ensembldb">ensembldb</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="load-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h1>
<p>The following data corresponds to a small subset of the dataset from <span class="citation">(Tasic Bosiljka 2018)</span> and is readily available from the <code>satuRn</code> package. To check how the subset was generate, please check <code><a href="../reference/Tasic_counts_vignette.html">?Tasic_counts_vignette</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="co"># transcript expression matrix</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">)</span> <span class="co"># metadata</span></code></pre></div>
</div>
<div id="data-pre-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#data-pre-processing" class="anchor"></a>Data pre-processing</h1>
<p>We start the analysis from scratch, in order to additionally showcase some of the prerequisite steps for performing a DTU analysis.</p>
</div>
<div id="import-transcript-information" class="section level1">
<h1 class="hasAnchor">
<a href="#import-transcript-information" class="anchor"></a>Import transcript information</h1>
<p>First, we need an object that links the transcripts the expression matrix to their corresponding genes. We suggest using <code>AnnotationHub</code> and <code>ensembldb</code> for this purpose.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># load the annotation resource.</span>
<span class="va">all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="st">"EnsDb"</span><span class="op">)</span> <span class="co"># query for all available EnsDb databases</span>
<span class="va">ahEdb</span> <span class="op">&lt;-</span> <span class="va">all</span><span class="op">[[</span><span class="st">"AH75036"</span><span class="op">]</span><span class="op">]</span> <span class="co"># for Mus musculus (choose correct release date)</span>
<span class="va">txs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-exonsBy.html">transcripts</a></span><span class="op">(</span><span class="va">ahEdb</span><span class="op">)</span></code></pre></div>
<div id="data-wrangling" class="section level2">
<h2 class="hasAnchor">
<a href="#data-wrangling" class="anchor"></a>Data wrangling</h2>
<p>Next, we perform some data wrangling steps to get the data in a format that is suited for satuRn. First, we create a <code>DataFrame</code> or <code>Matrix</code> linking transcripts to their corresponding genes. ! Important: <code>satuRn</code> is implemented such that the columns with transcript identifiers is names <code>isoform_id</code>, while the column containing gene identifiers should be named <code>gene_id</code>. In addition, following chunk removes transcripts that are the only isoform expressed of a certain gene, as they cannot be used in a DTU analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get the transcript information in correct format</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="cn">NA</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">txs</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">txInfo</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"isoform_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span>
<span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op">&lt;-</span> <span class="va">txs</span><span class="op">$</span><span class="va">tx_id</span>
<span class="va">txInfo</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="va">txs</span><span class="op">$</span><span class="va">gene_id</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">txInfo</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span>

<span class="co"># Remove transcripts that are the only isoform expressed of a certain gene</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span><span class="op">)</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">txInfo</span>, <span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span>, fromLast <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>

<span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
</div>
<div id="filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h2>
<p>Here we perform some feature-level filtering. For this task, we adopt the filtering criterium that is implemented in the R package <code>edgeR</code>. Alternatively, one could adopt the <code>dmFilter</code> criterium from the <code>DRIMSeq</code> R package, which provides a more stringent filtering when both methods are run in default settings. After filtering, we again remove transcripts that are the only isoform expressed of a certain gene.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filter_edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span>,
    design <span class="op">=</span> <span class="cn">NULL</span>,
    group <span class="op">=</span> <span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">brain_region</span>,
    lib.size <span class="op">=</span> <span class="cn">NULL</span>,
    min.count <span class="op">=</span> <span class="fl">10</span>,
    min.total.count <span class="op">=</span> <span class="fl">30</span>,
    large.n <span class="op">=</span> <span class="fl">20</span>,
    min.prop <span class="op">=</span> <span class="fl">0.7</span>
<span class="op">)</span> <span class="co"># more stringent than default to reduce run time of the vignette</span>

<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">filter_edgeR</span><span class="op">)</span></code></pre></div>
<pre><code>## filter_edgeR
## FALSE  TRUE 
##  5996 10982</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="va">filter_edgeR</span>, <span class="op">]</span>

<span class="co"># Update txInfo according to the filtering procedure</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>

<span class="co"># remove transcripts that are the only isoform expressed of a certain gene (after filtering)</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">txInfo</span>, <span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span>, fromLast <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span>

<span class="co"># satuRn requires the transcripts in the rowData and the transcripts in the count matrix to be in the same order.</span>
<span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span>, <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
</div>
<div id="create-a-design-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#create-a-design-matrix" class="anchor"></a>Create a design matrix</h2>
<p>Here we set up the design matrix of the experiment. The subset of the dataset from <span class="citation">(Tasic Bosiljka 2018)</span> contains cells of several different cell types (variable <code>cluster</code>) in two different areas of the mouse neocortex (variable <code>brain_region</code>). As such, we can model the data with a factorial design, i.e. by generating a new variable <code>group</code> that encompasses all different cell type - brain region combinations.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">brain_region</span>, <span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">cluster</span>, sep <span class="op">=</span> <span class="st">"."</span><span class="op">)</span></code></pre></div>
</div>
<div id="generate-summarizedexperiment" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-summarizedexperiment" class="anchor"></a>Generate SummarizedExperiment</h2>
<p>All three main functions of <code>satuRn</code> require a <code>SummarizedExperiment</code> object as an input class. See the SummarizedExperiment vignette <span class="citation">(Morgan Martin, n.d.)</span> for more information on this object class. Do not forget to include the design matrix formula (see above) to the SummarizedExperiment as indicated below. As such, the object contains all the information required for the downstream DTU analysis.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>
    assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">Tasic_counts_vignette</span><span class="op">)</span>,
    colData <span class="op">=</span> <span class="va">Tasic_metadata_vignette</span>,
    rowData <span class="op">=</span> <span class="va">txInfo</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html">metadata</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span><span class="op">)</span> <span class="co"># specify design formula from colData</span>
<span class="va">sumExp</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 9151 60 
## metadata(1): formula
## assays(1): counts
## rownames(9151): ENSMUST00000037739 ENSMUST00000228774 ...
##   ENSMUST00000127554 ENSMUST00000132683
## rowData names(2): isoform_id gene_id
## colnames(60): F2S4_160622_013_D01 F2S4_160624_023_C01 ...
##   F2S4_160919_010_B01 F2S4_160915_002_D01
## colData names(4): sample_name brain_region cluster group</code></pre>
</div>
</div>
<div id="fit-quasi-binomial-generalized-linear-models-models" class="section level1">
<h1 class="hasAnchor">
<a href="#fit-quasi-binomial-generalized-linear-models-models" class="anchor"></a>Fit quasi-binomial generalized linear models models</h1>
<p>The <code>fitDTU</code> function of <code>satuRn</code> is used to model transcript usage in different groups of samples or cells. Here we adopt the default settings of the function. Without parallel execution, this code runs for approximately 15 seconds on a 2018 macbook pro laptop.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.time</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "2021-01-20 20:48:02 CET"</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/fitDTU.html">fitDTU</a></span><span class="op">(</span>
    object <span class="op">=</span> <span class="va">sumExp</span>,
    formula <span class="op">=</span> <span class="op">~</span><span class="fl">0</span><span class="op">+</span><span class="va">group</span>, 
    parallel <span class="op">=</span> <span class="cn">FALSE</span>,
    BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html">bpparam</a></span><span class="op">(</span><span class="op">)</span>,
    verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.time</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "2021-01-20 20:48:18 CET"</code></pre>
<p>The resulting model fits are now saved into the <code>rowData</code> of our SummarizedExperiment object under the name <code>fitDTUModels</code>. These models can be accessed as follows:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUModels"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="st">"ENSMUST00000037739"</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUModels"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="co"># equivalent</span></code></pre></div>
<p>The models are instances of the <code>StatModel</code> class as defined in the <code>satuRn</code> package. These contain all relevant information for the downstream analysis. For more details, read the StatModel documentation with ?satuRn::<code>StatModel-class</code>.</p>
</div>
<div id="test-for-dtu" class="section level1">
<h1 class="hasAnchor">
<a href="#test-for-dtu" class="anchor"></a>Test for DTU</h1>
<p>Here we test for differential transcript usage between select groups of interest.</p>
<div id="set-up-contrast-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up-contrast-matrix" class="anchor"></a>Set up contrast matrix</h2>
<p>First, we set up a contrast matrix. This allows us to test for differential transcript usage between groups of interest. The <code>group</code> factor in this toy example contains three levels; (1) ALM.L5_IT_ALM_Tmem163_Dmrtb1, (2) ALM.L5_IT_ALM_Tnc, (3) VISp.L5_IT_VISp_Hsd11b1_Endou. Here we show to assess DTU between cells of the groups 1 and 3 and between cells of groups 2 and 3.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">group</span><span class="op">)</span>

<span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">0</span>, ncol <span class="op">=</span> <span class="fl">2</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span> <span class="co"># initialize contrast matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Contrast1"</span>, <span class="st">"Contrast2"</span><span class="op">)</span>

<span class="va">L</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span>, <span class="st">"ALM.L5_IT_ALM_Tnc"</span><span class="op">)</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>
<span class="va">L</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span>, <span class="st">"ALM.L5_IT_ALM_Tmem163_Dmrtb1"</span><span class="op">)</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>
<span class="va">L</span> <span class="co"># final contrast matrix</span></code></pre></div>
<pre><code>##                               Contrast1 Contrast2
## ALM.L5_IT_ALM_Tmem163_Dmrtb1          0        -1
## ALM.L5_IT_ALM_Tnc                    -1         0
## VISp.L5_IT_VISp_Hsd11b1_Endou         1         1</code></pre>
</div>
<div id="perform-the-test" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-the-test" class="anchor"></a>Perform the test</h2>
<p>Next we can perform differential usage testing using <code>testDTU</code>. We again adopt default settings. For more information on the parameter settings, please fitting the help file of the testDTU function.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/testDTU.html">testDTU</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sumExp</span>, 
                          contrasts <span class="op">=</span> <span class="va">L</span>, 
                          plot <span class="op">=</span> <span class="cn">FALSE</span>, 
                          sort <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The test results are now saved into the <code>rowData</code> of our SummarizedExperiment object under the name <code>fitDTUResult_</code> followed by the name of the contrast of interest (i.e. the column names of the contrast matrix). The results can be accessed as follows:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast1"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># first contrast</span></code></pre></div>
<pre><code>##                     estimates        se      df          t       pval
## ENSMUST00000037739 -0.5411265 0.4828721 59.4294 -1.1206416 0.26694865
## ENSMUST00000228774  0.5411265 0.4828721 59.4294  1.1206416 0.26694865
## ENSMUST00000025204  0.1929718 0.1952590 61.4294  0.9882864 0.32688926
## ENSMUST00000237499 -0.1929718 0.1952590 61.4294 -0.9882864 0.32688926
## ENSMUST00000042857 -0.8245461 0.4351069 58.4294 -1.8950427 0.06303775
## ENSMUST00000114415  0.8245461 0.4351069 58.4294  1.8950427 0.06303775
##                    regular_FDR empirical_pval empirical_FDR
## ENSMUST00000037739   0.6450007      0.3952020     0.9769609
## ENSMUST00000228774   0.6450007      0.4148973     0.9797565
## ENSMUST00000025204   0.6977858      0.4727593     0.9834579
## ENSMUST00000237499   0.6977858      0.4515026     0.9808231
## ENSMUST00000042857   0.3566360      0.1579658     0.9152296
## ENSMUST00000114415   0.3566360      0.1685028     0.9203772</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast2"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># second contrast</span></code></pre></div>
<pre><code>##                     estimates        se      df         t        pval
## ENSMUST00000037739 -0.3801339 0.4941514 59.4294 -0.769266 0.444782126
## ENSMUST00000228774  0.3801339 0.4941514 59.4294  0.769266 0.444782126
## ENSMUST00000025204  0.2971434 0.1921038 61.4294  1.546786 0.127051689
## ENSMUST00000237499 -0.2971434 0.1921038 61.4294 -1.546786 0.127051689
## ENSMUST00000042857 -1.4866500 0.4997583 58.4294 -2.974738 0.004257964
## ENSMUST00000114415  1.4866500 0.4997583 58.4294  2.974738 0.004257964
##                    regular_FDR empirical_pval empirical_FDR
## ENSMUST00000037739   0.7772316     0.56104440     0.9893602
## ENSMUST00000228774   0.7772316     0.58504064     0.9893602
## ENSMUST00000025204   0.5021149     0.26790898     0.9893602
## ENSMUST00000237499   0.5021149     0.25297843     0.9893602
## ENSMUST00000042857   0.1491038     0.03349528     0.9893602
## ENSMUST00000114415   0.1491038     0.03654226     0.9893602</code></pre>
</div>
</div>
<div id="visualize-dtu" class="section level1">
<h1 class="hasAnchor">
<a href="#visualize-dtu" class="anchor"></a>Visualize DTU</h1>
<p>Finally, we may visualize the usage of select transcripts in select groups of interest.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span><span class="op">]</span>
<span class="va">group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"ALM.L5_IT_ALM_Tnc"</span><span class="op">]</span>

<span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sumExp</span>, 
                         contrast <span class="op">=</span> <span class="st">"Contrast1"</span>, 
                         groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span>, 
                         coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>, 
                         summaryStat <span class="op">=</span> <span class="st">"model"</span>, 
                         transcripts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENSMUST00000081554"</span>, <span class="st">"ENSMUST00000195963"</span>, <span class="st">"ENSMUST00000132062"</span><span class="op">)</span>, 
                         genes <span class="op">=</span> <span class="cn">NULL</span>, 
                         top.n <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>

<span class="co"># to have same layout as in our paper</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"VISp"</span>, <span class="st">"ALM"</span><span class="op">)</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>, <span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html">scale_x_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Hsd11b1_Endou"</span>, <span class="st">"Tnc"</span><span class="op">)</span><span class="op">)</span>

    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<pre><code>## Warning: Removed 1 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="Vignette_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<pre><code>## Warning: Removed 1 rows containing non-finite values (stat_ydensity).

## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="Vignette_files/figure-html/unnamed-chunk-16-2.png" width="700"></p>
<pre><code>## Warning: Removed 1 rows containing non-finite values (stat_ydensity).

## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="Vignette_files/figure-html/unnamed-chunk-16-3.png" width="700"></p>
</div>
<div id="session" class="section level1">
<h1 class="hasAnchor">
<a href="#session" class="anchor"></a>Session</h1>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ggplot2_3.3.3               SummarizedExperiment_1.20.0
##  [3] MatrixGenerics_1.2.0        matrixStats_0.57.0         
##  [5] edgeR_3.32.1                limma_3.46.0               
##  [7] ensembldb_2.14.0            AnnotationFilter_1.14.0    
##  [9] GenomicFeatures_1.42.1      AnnotationDbi_1.52.0       
## [11] Biobase_2.50.0              GenomicRanges_1.42.0       
## [13] GenomeInfoDb_1.26.2         IRanges_2.24.1             
## [15] S4Vectors_0.28.1            AnnotationHub_2.22.0       
## [17] BiocFileCache_1.14.0        dbplyr_2.0.0               
## [19] BiocGenerics_0.36.0         satuRn_0.99.0              
## [21] knitr_1.30                 
## 
## loaded via a namespace (and not attached):
##  [1] ProtGenerics_1.22.0           bitops_1.0-6                 
##  [3] fs_1.5.0                      bit64_4.0.5                  
##  [5] progress_1.2.2                httr_1.4.2                   
##  [7] rprojroot_2.0.2               tools_4.0.2                  
##  [9] R6_2.5.0                      lazyeval_0.2.2               
## [11] DBI_1.1.1                     colorspace_2.0-0             
## [13] withr_2.4.0                   prettyunits_1.1.1            
## [15] tidyselect_1.1.0              bit_4.0.4                    
## [17] curl_4.3                      compiler_4.0.2               
## [19] textshaping_0.2.1             xml2_1.3.2                   
## [21] desc_1.2.0                    DelayedArray_0.16.0          
## [23] labeling_0.4.2                rtracklayer_1.50.0           
## [25] scales_1.1.1                  pbapply_1.4-3                
## [27] askpass_1.1                   rappdirs_0.3.1               
## [29] pkgdown_1.6.1                 Rsamtools_2.6.0              
## [31] systemfonts_0.3.2             stringr_1.4.0                
## [33] digest_0.6.27                 rmarkdown_2.6                
## [35] XVector_0.30.0                pkgconfig_2.0.3              
## [37] htmltools_0.5.1               fastmap_1.0.1                
## [39] rlang_0.4.10                  rstudioapi_0.13              
## [41] RSQLite_2.2.2                 shiny_1.5.0                  
## [43] farver_2.0.3                  generics_0.1.0               
## [45] BiocParallel_1.24.1           dplyr_1.0.3                  
## [47] RCurl_1.98-1.2                magrittr_2.0.1               
## [49] GenomeInfoDbData_1.2.4        Matrix_1.2-18                
## [51] Rcpp_1.0.6                    munsell_0.5.0                
## [53] lifecycle_0.2.0               stringi_1.5.3                
## [55] yaml_2.2.1                    zlibbioc_1.36.0              
## [57] grid_4.0.2                    blob_1.2.1                   
## [59] promises_1.1.1                crayon_1.3.4                 
## [61] lattice_0.20-41               splines_4.0.2                
## [63] Biostrings_2.58.0             hms_1.0.0                    
## [65] locfit_1.5-9.4                pillar_1.4.7                 
## [67] boot_1.3-25                   biomaRt_2.46.0               
## [69] XML_3.99-0.5                  glue_1.4.2                   
## [71] BiocVersion_3.12.0            evaluate_0.14                
## [73] BiocManager_1.30.10           vctrs_0.3.6                  
## [75] httpuv_1.5.5                  locfdr_1.1-8                 
## [77] openssl_1.4.3                 gtable_0.3.0                 
## [79] purrr_0.3.4                   assertthat_0.2.1             
## [81] xfun_0.19                     mime_0.9                     
## [83] xtable_1.8-4                  later_1.1.0.1                
## [85] ragg_0.4.0                    tibble_3.0.5                 
## [87] GenomicAlignments_1.26.0      memoise_1.1.0                
## [89] ellipsis_0.3.1                interactiveDisplayBase_1.28.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-SummarizedExperiment" class="csl-entry">
Morgan Martin, Hester Jim, Obenchain Valerie. n.d. <span>“<span>SummarizedExperiment</span>.”</span> <a href="https://bioconductor.org/packages/SummarizedExperiment">https://bioconductor.org/packages/SummarizedExperiment</a>.
</div>
<div id="ref-Tasic2018" class="csl-entry">
Tasic Bosiljka, Lucas T. Graybuck, Zizhen Yao. 2018. <span>“<span class="nocase">Shared and distinct transcriptomic cell types across neocortical areas</span>.”</span> <em>Nature</em>, 72–78. https://doi.org/<a href="https://doi.org/10.1038/s41586-018-0654-5">https://doi.org/10.1038/s41586-018-0654-5</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jeroen Gilis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
